File: Interviews/LeetCode/Topic-Wise/Binary-Search/Easy/Guess-Number-Higher-or-Lower/374.cpp
class Solution {
   public:
    int guessNumber(int n) {
        int lo = 1, hi = n;
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            int slope = guess(mid);
            if (slope == 0) {
                return mid;
            } else if (slope == -1) {
                hi = mid - 1;
            } else {
                lo = mid + 1;
            }
        }

        return lo;
    }
};

File: Interviews/LeetCode/Topic-Wise/Binary-Search/Easy/Guess-Number-Higher-or-Lower/374.py
class Solution:
    def guessNumber(self, n: int) -> int:
        lo, hi = 1, n
        while lo <= hi:
            mid = lo + (hi - lo) // 2
            slope = guess(mid)
            if slope == 0:
                return mid
            elif slope == -1:
                hi = mid - 1
            else:
                lo = mid + 1
        return lo
